realm/realm-library/src/main/java/io/realm/RealmObject.java

```diff
@@ -32,16 +32,7 @@
  * In Realm you define your RealmObject classes by sub-classing RealmObject and adding fields to be persisted. You then 
  * create your objects within a Realm, and use your custom subclasses instead of using the RealmObject class directly.
  * <p>
- * An annotation processor will create a proxy class for your RealmObject subclass. The getters and setters should not
- * contain any custom code of logic as they are overridden as part of the annotation process.
- * <p>
- * A RealmObject is currently limited to the following:
- *
- * <ul>
- *   <li>Private fields.</li>
- *   <li>Getter and setters for these fields.</li>
- *   <li>Static methods.</li>
- * </ul>
+ * An annotation processor will create a proxy class for your RealmObject subclass.
```

Comment:
Perhaps just remove this? It is just an implementation detail that doesn't really impact the user.